packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts

```diff
@@ -60,22 +61,114 @@ export default class MuiDetailGenerator implements DetailGenerator {
         );
 
         if (widgetParentNode) {
-          let formikDeclarationNode = this.findGridDeclaration(widgetParentNode);
+          let formikDeclarationNode =
+            this.findGridDeclaration(widgetParentNode);
 
           if (formikDeclarationNode) {
-            let propertyAssigmentArray: ts.PropertyAssignment[] = []; 
-            findPropertyAssignment(formikDeclarationNode.getChildAt(1), propertyAssigmentArray);
+            let propertyAssigmentArray: ts.PropertyAssignment[] = [];
+            findPropertyAssignment(
+              formikDeclarationNode.getChildAt(1),
+              propertyAssigmentArray
+            );
+
+            if (propertyAssigmentArray) {
+              ast = this.addNewField(
+                formikDeclarationNode,
+                propertyAssigmentArray,
+                property,
+                ast
+              );
+            }
+          }
+        }
+
+        let gridElements: ts.JsxElement[] = [];
+        //Add grid with field
+        if (widgetParentNode) {
+          this.findGridElement(widgetParentNode, gridElements);
+        }
 
-             if (propertyAssigmentArray) {
-               ast = this.addNewField(formikDeclarationNode, propertyAssigmentArray, property, ast);
-             }
+        if (gridElements) {
+          if (widgetParentNode) {
+            let gridContainers: ts.JsxElement[] = [];
+            this.findGridContainer(widgetParentNode, gridContainers);
+            ast = this.addNewGridElement(
+              gridElements,
+              gridContainers[0],
+              property,
+              ast
+            );
           }
         }
 
         alteredSource = this.printSourceCode(ast);
         console.log(alteredSource);
       }
     }
+
+    return alteredSource
+  }
+
+  findGridElement(parentNode: ts.Node, array: ts.JsxElement[]) {
+    if (parentNode != undefined) {
+      if (parentNode.getChildCount() > 0) {
+        var children = parentNode.getChildren();
+        children.forEach((child) => {
+          if (ts.isJsxElement(child)) {
+            if (child.getFullText().startsWith("Grid item", 1)) {
+              array.push(child);
+            } else {
+              this.findGridElement(child, array);
+            }
+          } else {
+            this.findGridElement(child, array);
+          }
+        });
+      }
+    }
+  }
+  
+  findGridContainer(parentNode: ts.Node, array: ts.JsxElement[]) {
```

Comment:
please, use proper parameter naming (`array` is too generic name)

Reply:
fixed